prov/gni/src/gnix_ep.c

```diff
@@ -1167,21 +1162,17 @@ static inline struct gnix_fab_req *__find_tx_req(
 	GNIX_DEBUG(FI_LOG_EP_CTRL, "searching VCs for the correct context to"
 			" cancel, context=%p", context);
 
-	fastlock_acquire(&ep->vc_list_lock);
-	dlist_for_each(&ep->wc_vc_list, vc, entry)
-	{
-		GNIX_DEBUG(FI_LOG_EP_CTRL, "checking vc=%p\n", vc);
+	while ((vc = _gnix_nic_next_pending_vc(ep->nic))) {
```

Comment:
If a request is in vc->tx_queue it hasn't been successfully initiated yet.  So any requests on any VC's tx_queue should be able to be canceled.  There are situations where a VC is not scheduled but has non-empty tx_queue (we're waiting for TXDs to be freed, at which time the VC should be scheduled).